src/online/initialization.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"net/http"
+	"vaas/structs"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+)
+
+func newSharedGame(c *gin.Context) {
+	/*
+		This API method takes in a metadata struct in the request body, sends it to Engine to make a new game, creates a SharedGame struct, and sends it to Mongo to store
+
+		@param: metadata for new game (structs.Metadata)
+		@return: new shared game created (structs.SharedGame)
+	*/
+	newMetadata := structs.GameMetadata{}
+	if err := c.ShouldBindJSON(&newMetadata); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Bad request format, expected {wordLength: int, maxGuesses: int}"})
+		return
+	}
+
+	// Send metadata to engine to create new game
+	newGame, err := engine_newGame(newMetadata)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, structs.Message{Message: "Engine newGame issue: " + err.Error()})
+	}
+
+	/*
+		Create shared game structure. The game ID and word received in Engine response are put into the 'Games' map and 'Word' field, respectively
+	*/
+	sharedGame := structs.SharedGame{}
+	sharedGame.SharedGameID = uuid.NewString()
+	sharedGame.HostID = newMetadata.UserId
+	sharedGame.Games = make(map[string]string)
+	sharedGame.Games[newMetadata.UserId] = newGame.Metadata.GameID
+	sharedGame.State = "waiting"
+	sharedGame.WinnerID = ""
+	sharedGame.Word = newGame.Word
+
+	// Send shared game to Mongo communication method
+	mongo_response, err := mongo_createSharedGame(sharedGame)
+
+	// Respond based on Mongo response
+	if err != nil || mongo_response == "Could not create shared game due to Mongo error" {
```

Comment:
```suggestion
	if err = mongo_createSharedGame(sharedGame); err != nil {
```
We can just keep this `err != nil` instead of having the additional `|| mongo_response == "Could not create shared game due to Mongo error"`

(See comment at the end of mongo_createSharedGame to see how to simplify this)